hat/assets/js/apps/Iaso/domains/app/contexts/SideBarContext.tsx

```diff
@@ -0,0 +1,33 @@
+import React, {
+    createContext,
+    useCallback,
+    useContext,
+    useMemo,
+    useState,
+} from 'react';
+
+const SidebarContext = createContext({
+    isOpen: false,
+    toggleSidebar: () => {
+        /* noop */
+    },
+});
+
+export const useSidebar = () => useContext(SidebarContext);
+
+export const SidebarProvider = ({ children }) => {
+    const [isOpen, setIsOpen] = useState(false);
+    const toggleSidebar = useCallback(() => setIsOpen(!isOpen), [isOpen]);
```

Comment:
remember that one 😉?

```suggestion
    const toggleSidebar = () => setIsOpen(value => !value);
```